oauthdoctor/oauth/installed_app_test.go

```diff
@@ -1,34 +1,125 @@
 package oauth
 
 import (
-  "strings"
-  "testing"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/googleads/google-ads-doctor/oauthdoctor/diag"
+	"golang.org/x/oauth2"
 )
 
+func setupFakeOAuthServer() (*httptest.Server, func()) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/auth", func(w http.ResponseWriter, r *http.Request) {
+		// Should return authorization code back to the user
+		w.Write([]byte("code=mockauthcode"))
```

Comment:
nit: this is a fake not a mock server, so do you want to call this a mockoauthcode?

Fake: code that implements an interface but contains fixed data and no logic. Simply returns "good" or "bad" data depending on the implementation.

Mock: code that implements an interface and allows the ability to dynamically set the values to return/exceptions to throw from particular methods and provides the ability to check if particular methods have been called/not called.

Reply:
Fixed